### ISSUE TITLE
Elements in treeviews are not counted in Eclipse IDE 2018‑12

### ISSUE THREAD

**Issue by taymun26** [nvaccess/nvda#9127]
### Actual behavior:
When using Eclipse IDE 2018‑12 (only 64 Bit available) with NVDA items in a treeview are not counted as expected.
1. Open Eclipse
2. Press ctrl+n to open the new dialogue.
3. Press Tab. NVDA says "one of one" although there are dozens of elements in the treeview. When opening the nested elements with the right arrow key NVDA cannot count the elements either. Same  in the Eclipse's preference dialogue.

### Expected behavior:
NVDA does announce the correct count of elements in the treeview as it used to be in former Eclipse versions. Probably the change from 32 bit to 64 bit Eclipse may have an impact.
From the current Eclipse release, only 64 bit (x86_64) is available.

#### NVDA installed/portable/running from source:
Both tested. Current and former NVDA versions.

#### NVDA version:
2018.4, 2018.3, 2017.4

#### Windows version:
Windows 10 1709

#### Name and version of other software in use when reproducing the issue:
Eclipse IDE 2018‑12, JDK 11.01, both 64 Bit

#### Does the issue still occur after restarting your PC?
Not tested.

#### Have you tried any other versions of NVDA?
Yes

**Comment by Adriani90** [comment]
@ABuffEr maybe your addon "columns Viewer" could solve this problem. If yes, then maybe you could raise a Pull Request against NVDA to integrate a solution for this problem?

**Comment by josephsl** [comment]
Hi, I beg to differ: columns review might not be the solution; rather, it may have to do with Java Access Bridge. Thanks.

**Comment by ABuffEr** [comment]
Hi,
as Joseph, I think ColumnsReview cannot help here.
I have no time to test with Eclipse today, but, if I understand the problem, consider that a very similar or identical issue occurs on my system (Windows 10 64-bit) opening regedit from run dialog and expanding the tree. And I'm quite sure that it's a regression, because I remember to have noticed it, then it was disappeared, and, some weeks ago, it appeared again. Sorry for vagueness... could exist an open/closed issue too.

**Comment by Adriani90** [comment]
the same issue occurs in CCleaner in the tree view for example.

**Comment by Adriani90** [comment]
cc: @michaelDCurran, @derekriemer, @feerrenrut

**Comment by lukaszgo1** [comment]
Probably a duplicate of #7041.

**Comment by Adriani90** [comment]
I suggest to let this open and ask the reporter later if the problem is resolved after the referenced issue has been solved. The reason is that we handle different situations and it is not clear if they are completely the same. However, I would continue the discussion in the referenced issue because it goes into more detail.

**Comment by taymun26** [comment]
This seems to be an inter-process-communication issue as stated in #7041 .
In Eclipse 4.9, 32 Bit this works fine. 32 Bit support is dropped in Eclipse 4.10 and up
(https://www.eclipse.org/eclipse/news/4.10/platform.php#java32-removal).
Eclipse /NVDA should not require the Java Access Bridge since Eclipse is based on SWT, i.e. platform-native controls.

Here is some excerpt from Eclipse 4.10 and 4.9 in the new dialogue:
Eclipse 4.10, 64 bit, not working:
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (08:31:23.868):
Developer info for navigator object:
name: u'Java Project'
role: ROLE_TREEVIEWITEM
states: STATE_INVISIBLE, STATE_OFFSCREEN, STATE_FOCUSED, STATE_SELECTED, STATE_FOCUSABLE, STATE_SELECTABLE
isFocusable: True
hasFocus: True
Python object: <baseObject.Dynamic_TreeViewItemOutlineItemIAccessible object at 0x07EDBBB0>
Python class mro: (<class 'baseObject.Dynamic_TreeViewItemOutlineItemIAccessible'>, <class 'NVDAObjects.IAccessible.sysTreeView32.TreeViewItem'>, <class 'NVDAObjects.IAccessible.OutlineItem'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: RectLTWH(left=234, top=362, width=82, height=23)
value: None
appModule: <'eclipse' (appName u'eclipse', process ID 11944) at address 7f3b210>
appModule.productName: u'OpenJDK Platform 11'
appModule.productVersion: u'11.0.1'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 69068
windowClassName: u'SysTreeView32'
windowControlID: 69068
windowStyle: 1445020213
windowThreadID: 13084
windowText: u''
displayText: u'ClassEnum'
IAccessibleObject: <POINTER(IAccessible) ptr=0x9708120 at 803fdf0>
IAccessibleChildID: 29
IAccessible event parameters: windowHandle=69068, objectID=-4, childID=29
IAccessible accName: u'Java Project'
IAccessible accRole: ROLE_SYSTEM_OUTLINEITEM
IAccessible accState: STATE_SYSTEM_SELECTABLE, STATE_SYSTEM_OFFSCREEN, STATE_SYSTEM_SELECTED, STATE_SYSTEM_INVISIBLE, STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (3244038)
IAccessible accDescription: None
IAccessible accValue: u'1'


Eclipse 4.9 32 bit working:
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (08:34:00.908):
Developer info for navigator object:
name: u'General'
role: ROLE_TREEVIEWITEM
states: STATE_COLLAPSED, STATE_FOCUSABLE, STATE_FOCUSED, STATE_SELECTED, STATE_SELECTABLE
isFocusable: True
hasFocus: True
Python object: <baseObject.Dynamic_TreeViewItemOutlineItemIAccessible object at 0x07EC8570>
Python class mro: (<class 'baseObject.Dynamic_TreeViewItemOutlineItemIAccessible'>, <class 'NVDAObjects.IAccessible.sysTreeView32.TreeViewItem'>, <class 'NVDAObjects.IAccessible.OutlineItem'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: RectLTWH(left=417, top=440, width=55, height=23)
value: None
appModule: <'eclipse' (appName u'eclipse', process ID 11004) at address 7f3b890>
appModule.productName: u'Java(TM) Platform SE 8'
appModule.productVersion: u'8.0.2020.3'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 134604
windowClassName: u'SysTreeView32'
windowControlID: 134604
windowStyle: 1445020213
windowThreadID: 10184
windowText: u''
displayText: u'General'
IAccessibleObject: <POINTER(IAccessible) ptr=0xa28c250 at 7ec0580>
IAccessibleChildID: 1
IAccessible event parameters: windowHandle=134604, objectID=-4, childID=1
IAccessible accName: u'General'
IAccessible accRole: ROLE_SYSTEM_OUTLINEITEM
IAccessible accState: STATE_SYSTEM_SELECTABLE, STATE_SYSTEM_SELECTED, STATE_SYSTEM_COLLAPSED, STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (3146758)
IAccessible accDescription: None
IAccessible accValue: u'0'

**Comment by Adriani90** [comment]
@shadyar, @taymun26, are you still having this issue with Eclipse 2024.06 and NVDA 2024.2 Stable or 2024.3 Beta?

**Comment by shadyar** [comment]
It seems that the issue has been resolved with NVDA v24.2
